### PR TITLE
support github actions as a CI provider

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

Fixes https://github.com/conda-forge/conda-smithy/issues/1361

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -609,6 +609,11 @@ def _render_ci_provider(
             config,
             forge_config,
         )
+        for channel_target in migrated_combined_variant_spec.get("channel_targets", []):
+            if channel_target.startswith("conda-forge ") and provider_name == "github_actions":
+                raise RuntimeError("Using github_actions as the CI provider inside "
+                                   "conda-forge github org is not allowed as github actions "
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:615]:
```suggestion
                                   "conda-forge github org is not allowed in order "
```

**Review by beckermr — COMMENTED** [review]
One comment on an error string. Do we want to adjust the secrets code so that smithy can add the binstar token to the github repo secrets?

**Comment by isuruf** [comment]
> Do we want to adjust the secrets code so that smithy can add the binstar token to the github repo secrets?

We could, but I'd prefer not to do that for now and wait for contributions from others.

**Review by beckermr — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Isuru for the PR and Matt for the review! 😄

**Comment by jaimergp** [comment]
We could also add the logging groups support specific to GH Actions here. I'll work on that as part of the fix for https://github.com/conda-forge/conda-smithy/pull/1436#issuecomment-791040414

**Pull request merged**